src/dnaio/_core.pyx

```diff
@@ -423,30 +457,42 @@ def fastq_iter(file, sequence_class, Py_ssize_t buffer_size):
             buf[0:bufstart] = buf[record_start:bufend]
 
 
-def record_names_match(header1: str, header2: str):
+def record_names_match(header1: Union[str, bytes], header2: Union[str, bytes]):
```

Comment:
I haven’t fully made up my mind here, but spontaneously, I’m not very fond of this solution and would probably prefer to have a separate function `record_names_match_bytes`.

It would probably make sense to turn this into a method on `Sequence` and `BytesSequence` (but that needs a separate PR).

Reply:
A `id_matches` method seems like a good idea for a later PR indeed.

Reply:
I implemented a separate name matcher. I think the difference between `record_names_match` and `record_names_match_bytes` shows why I desparately want a BytesSequence object. 

This did however require some extra python code in the paired readers. I made sure the two commits are atomic, so they can easily be converted if you like the previous solution better.